### PR TITLE
Elisa/6062 patient query param addition

### PR DESCRIPTION
**Pull request by emyl3**
See #6128 for details
This PR was opened to ensure backward compatibility with an old frontend that did not make use of the new `archivedStatus` parameter.

patients with `includeArchived`
```
patients(
      facilityId: $facilityId
      pageNumber: 0
      pageSize: 100
      includeArchived: $includeArchived
      namePrefixMatch: $namePrefixMatch
      includeArchivedFacilities: $includeArchivedFacilities
    ) {
      internalId
      firstName
      lastName
      middleName
      birthDate
      gender
      telephone
      email
      emails
      phoneNumbers {
        type
        number
      }
    }
```

patients with `archivedStatus`
```
patients(
      facilityId: $facilityId
      pageNumber: 0
      pageSize: 100
      archivedStatus: $archivedStatus
      namePrefixMatch: $namePrefixMatch
      includeArchivedFacilities: $includeArchivedFacilities
    ) {
      internalId
      firstName
      lastName
      middleName
      birthDate
      gender
      telephone
      email
      emails
      phoneNumbers {
        type
        number
      }
    }
    
```

patientsCount with `includeArchived`
```
    patientsCount(
      facilityId: $facilityId
      includedArchived: $includeArchived
      namePrefixMatch: $namePrefixMatch
    )  
```

patientsCount with `archivedStatus`
```
    patientsCount(
      facilityId: $facilityId
      archivedStatus: $archivedStatus
      namePrefixMatch: $namePrefixMatch
    )  
```

Second PR to deprecate and clean up old parameter usage:

### CODE REVIEW

File: backend/src/main/java/gov/cdc/usds/simplereport/api/patient/PatientResolver.java [hunk 1]
```diff
@@ -26,30 +27,40 @@ public PatientResolver(PersonService ps, OrganizationService os) {
...
   // authorization happens in calls to PersonService
   @QueryMapping
   public long patientsCount(
       @Argument UUID facilityId,
-      @Argument boolean includeArchived,
+      @Argument Boolean includeArchived,
+      @Argument ArchivedStatus archivedStatus,
       @Argument String namePrefixMatch) {
-    return _ps.getPatientsCount(facilityId, includeArchived, namePrefixMatch, false);
+    ArchivedStatus status;
```
Comment (emyl3) [inline backend/src/main/java/gov/cdc/usds/simplereport/api/patient/PatientResolver.java:57]: ⚠️ for backward compatibility -- will remove in separate PR once this one is deployed

File: backend/src/main/java/gov/cdc/usds/simplereport/api/patient/PatientResolver.java [hunk 2]
```diff
@@ -26,30 +27,40 @@ public PatientResolver(PersonService ps, OrganizationService os) {
...
   public List<Person> patients(
       @Argument UUID facilityId,
       @Argument int pageNumber,
       @Argument int pageSize,
-      @Argument boolean includeArchived,
+      @Argument Boolean includeArchived,
+      @Argument ArchivedStatus archivedStatus,
       @Argument String namePrefixMatch,
       @Argument boolean includeArchivedFacilities) {
+    ArchivedStatus status;
```
Comment (emyl3) [inline backend/src/main/java/gov/cdc/usds/simplereport/api/patient/PatientResolver.java:40]: ⚠️ for backward compatibility -- will remove in separate PR once this one is deployed

File: backend/src/main/resources/graphql/main.graphqls [hunk 3]
```diff
@@ -383,6 +383,8 @@ type Query {
     pageSize: Int = 5000
     includeArchived: Boolean = false
```
Comment (emyl3) [inline backend/src/main/resources/graphql/main.graphqls:384]: ⚠️ for backward compatibility -- will deprecate in separate PR once this one is deployed

File: backend/src/main/resources/graphql/main.graphqls [hunk 4]
```diff
@@ -392,6 +394,8 @@ type Query {
     facilityId: ID
     includeArchived: Boolean = false
```
Comment (emyl3) [inline backend/src/main/resources/graphql/main.graphqls:395]: ⚠️ for backward compatibility -- will deprecate in separate PR once this one is deployed

File: backend/src/main/java/gov/cdc/usds/simplereport/api/patient/PatientResolver.java [hunk 5]
```diff
@@ -25,31 +26,42 @@ public PatientResolver(PersonService ps, OrganizationService os) {
     _os = os;
   }
 
+  private ArchivedStatus getArchivedStatus(Boolean includeArchived, ArchivedStatus archivedStatus) {
```
Comment (emyl3) [inline backend/src/main/java/gov/cdc/usds/simplereport/api/patient/PatientResolver.java:29]: ⚠️ for backward compatibility -- will remove in separate PR once this one is deployed

**Review by fzhao99 — APPROVED** [review]

**Review by johanna-skylight — APPROVED** [review]
LGTM!

**Pull request merged**